### PR TITLE
Only the first printer is reported after discovery

### PR DESCRIPTION
**Pull request by JesperBadstue**
I noticed that when discovering printers the `DiscoveryListener` was first called with one printer, and shortly thereafter with two printers etc by the Eposon lib.
The problem is then that as soon at we get a printer in the listener we send the event `onDiscoveryDone` and the react part of the lib stops listening to any more events. This means that the second discovered printer was never reported.
I have added a `mOnDiscoveryReporter ` that delays the emitting of `onDiscoveryDone` for 3 seconds to allow us to discover more than one printer. The 3 seconds is more or less arbitrarily chosen and I guess it could be made configurable if needed.

### CODE REVIEW

**Comment by tr3v3r** [comment]
Hi @JesperBadstue ! Is this issue also reproduced on iOS? 
Since I don't have several devices, could please try to search printers by running an example app from this repo?

Actually, I suppose it's not an ideal solution for this issue, since are already waiting for 5 seconds after running Discovery.start(). So possibly better to shift sending an event into performing the discovery method?

Please look through my notes in the code

File: android/src/main/java/com/reactnativeescposprinter/EscPosPrinterDiscoveryModule.java [hunk 1]
```diff
@@ -216,32 +217,39 @@ public void onDiscovery(final DeviceInfo deviceInfo) {
...
+        }
+      });
 
-            if (mExtractUsbSerialNumber && usbAddress != "") {
-              String usbSerialNumber = getUsbSerialNumber(usbAddress);
-              printerData.putString("usbSerialNumber", usbSerialNumber);
+      if (mOnDiscoveryReporter == null) {
+        mOnDiscoveryReporter = new Runnable() {
+          public synchronized void run() {
+            WritableArray stringArray = Arguments.createArray();
```
Comment (tr3v3r) [inline android/src/main/java/com/reactnativeescposprinter/EscPosPrinterDiscoveryModule.java:226]: let's extract this forming and sending data into separate method like: 
```
public void sendDiscoveredDataToJS() {

  WritableArray stringArray = Arguments.createArray();
  for (int counter = 0; counter < mPrinterList.size(); counter++) {
      final DeviceInfo info = mPrinterList.get(counter);
      WritableMap printerData = Arguments.createMap();
      
     ........
     
     sendEvent(reactContext, "onDiscoveryDone", stringArray);
     }
```

And try to send in right after discovery done (after 5 sec) here:

```
  private void performDiscovery(MyCallbackInterface callback) {  
    final Handler handler = new Handler();
    final Runnable r = new Runnable() { 
         public void run() {   
           if(mPrinterList.size() > 0) {
                sendDiscoveredDataToJS() // will be invoked after 5 sec with acc. results
           }
           stopDiscovery();       
           callback.onDone("Search completed"); 
           }   
        };
      handler.postDelayed(r, 5000);
     }
```

**Review by tr3v3r — CHANGES_REQUESTED** [review]

**Comment by JesperBadstue** [comment]
> Hi @JesperBadstue ! Is this issue also reproduced on iOS?
> Since I don't have several devices, could please try to search printers by running an example app from this repo?
> 
> Actually, I suppose it's not an ideal solution for this issue, since we are already waiting for 5 seconds after running Discovery.start(). So possibly better to shift sending an event into performDiscovery method?
> 
> Please look through my notes in the code

Hi @tr3v3r
Unfortunately I dont have any iOS devices so I dont know if it has the same problems:(

You are right that we should just the 5 sec delay already in the `performDiscovery` I'll update the PR.

**Review by tr3v3r — APPROVED** [review]

**Comment by tr3v3r** [comment]
@JesperBadstue merged. I'll try to prepare a new release today. Thanks for your help!

**Pull request merged**